### DISCUSSION TITLE
QTime emit current view (hours, minutes, ...)

### DISCUSSION THREAD

**Discussion by 54mu3l** [quasarframework/quasar#13874] [Ideas / Proposals]
Currently there is no way to know in what "view-state" QTime is (hours, minutes, ...).

It would be great if QTime would either emit the current view state through the `details` object (`@update:model-value -> function(value, details)`)
or
if there would be a new event `@viewChanged-> function(view-state)` (maybe see also this discussion: https://github.com/quasarframework/quasar/discussions/12187)

---

Based on the change of QTime emit behavior (Quasar V1 vs. Quasar V2)
(see more here https://github.com/quasarframework/quasar/issues/13562#issuecomment-1169799907 and here https://github.com/quasarframework/quasar/issues/13872#issuecomment-1174794819) this is a must have. Since the suggested use of button slots (https://github.com/quasarframework/quasar/issues/13562#issuecomment-1169799907) isn't the UX we're going for. Thus, we need to know the current view-state.

---

Edit (10mins after creating this discussion):
Funny, seems like someone else has the same use case and just started a similar disucssion _at the same time_: https://github.com/quasarframework/quasar/discussions/13873

**Comment by flopail** [comment]
Yes, the same idea, it's me. Guilty! :D I also created the same bug issue (sorry for that), that's why I closed it: https://github.com/quasarframework/quasar/issues/13872

**Comment by 54mu3l** [comment]
Any thoughts/news on this?

Let me recap for you, what this is all about:

A new event that emits the current QTime view state would be really helpful for many use cases. I'm thinking of something like this: `@viewChanged-> function(view-state)`

---

Or as alternative (if we don't want to add a new event) we could also change the existing event `@update:model-value`.
`@update:model-value` currently emits `value` and `details` with the details looking something like this:
```
{
  year: 2022,
  month: 10,
  day: 28,
  hour: 19,
  minute: 36,
  second: null,
  millisecond: null,
  changed : true
}
```

It would be really helpful if `@update:model-value` could also emit the current view state, like this:
```
{
  year: 2022,
  month: 10,
  day: 28,
  hour: 19,
  minute: 36,
  second: null,
  millisecond: null,
  changed : true,
  viewState: 'hour'
}
```

Or if it's easier something like this would also help:
```
{
  year: 2022,
  month: 10,
  day: 28,
  hour: 19,
  minute: 36,
  second: null,
  millisecond: null,
  changed : true,
  hoursChanged: false,
  minutesChanged: true,
  meridiemChanged: false,
}
```

Thanks!
